pkg/agent/proxy/proxier.go

```diff
@@ -613,6 +628,27 @@ func (p *proxier) installServices() {
 	}
 }
 
+func (p *proxier) getMCSExportedServiceInfo(serviceCIDRIPv4 *net.IPNet, endpointIP string, svcPortName k8sproxy.ServicePortName) *openflowtypes.ServiceGroupInfo {
+	var mcsLocalService *openflowtypes.ServiceGroupInfo
+	// When the Endpoint is a local ClusterIP, it means the corresponding local Service is a member of the Multi-cluster Service.
+	if serviceCIDRIPv4 != nil && serviceCIDRIPv4.Contains(net.ParseIP(endpointIP)) {
+		mcsLocalService = &openflowtypes.ServiceGroupInfo{
+			ClusterIP: net.ParseIP(endpointIP),
+		}
+	}
+	// For any Multi-cluster Service, its name will be a combination with prefix `antrea-mc-` and
+	// exported Service's name. So we need to remove the prefix to look up the exported Service.
+	if mcsLocalService != nil && strings.HasPrefix(svcPortName.Name, mcServiceNamePrefix) {
+		exportedSvcPortName := svcPortName
+		exportedSvcPortName.Name = strings.TrimPrefix(svcPortName.Name, mcServiceNamePrefix)
+		if _, ok := p.serviceMap[exportedSvcPortName]; ok {
+			mcsLocalService.GroupID = p.groupCounter.AllocateIfNotExist(exportedSvcPortName, false)
```

Comment:
If all Endpoints of a local Service are removed but the Service is not removed yet, do we support but removing the bucket from the OF group which is used for the related multi-cluster Service?

Reply:
Yes, MC controller will be responsible to remove the Endpoint of MC Service if exported Service has no available Endpoints anymore. After the Endpoint is removed, the bucket will be removed because of Endpoint change.

Reply:
When the number of Endpoints of a Service is changed from N to 0, the group is still there and all buckets were removed.